griptape/drivers/vector/pgvector_vector_store_driver.py

```diff
@@ -0,0 +1,170 @@
+import uuid
+from typing import Optional
+from attr import define, field
+from dataclasses import dataclass
+from griptape.drivers import BaseVectorStoreDriver
+from sqlalchemy.engine import Engine
+from sqlalchemy import create_engine, Column, String, JSON
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Session
+from pgvector.sqlalchemy import Vector
+
+
+def vector_model_factory(table_name="griptape_vectors"):
+    Base = declarative_base()
+
+    @dataclass
+    class VectorModel(Base):
+        __tablename__ = table_name
+
+        id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, unique=True, nullable=False)
+        vector = Column(Vector())
+        namespace = Column(String)
+        meta = Column(JSON)
+
+    return VectorModel
+
+
+@define
+class PgVectorVectorStoreDriver(BaseVectorStoreDriver):
+    """A vector store driver to Postgres using the PGVector extension.
+
+    Attributes:
+        connection_string: The connection string for the Postgres database instance.
+        create_engine_params: Additional configuration params passed when creating the database connection.
+        engine: An optional sqlalchemy Postgres engine to use.
+    """
+
+    connection_string: str = field(default=None, kw_only=True)
+    create_engine_params: Optional[dict] = field(factory=dict, kw_only=True)
+    _Model: any = field(default=vector_model_factory, init=False)
+    engine: Optional[Engine] = field(default=None)
+
+    def initialize(
```

Comment:
Can we rename this (may be to `setup`?), so it doesn't sound too similar to `__init__`?

Reply:
Add type hints to all props and the return type.